File: readability/readability.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>
#include <ctype.h>


int get_grade(string s)
{
    int grade = 1;
    int l = 0, sents = 0, w = 1;
    int i = 0;
    float L = 0.0, S = 0.0;
    char c;

    while((c = s[i]) != '\0')
    {
        if (c == ' ')
        {
            w += 1; // spaces delimit words
        }
        else if (isalpha(c))
        {
            l++;
        }
        else if (c == '.' || c == '!' || c == '?')
        {
            sents += 1; //sentences are delimited by 3 chars
        }

        i++;    // avoid the infinite! :)
    }

    printf("%d words\n", w);
    printf("%d sents\n", sents);
    printf("%d letss\n", l);


    L = 100 * (float)l / w;
    S = 100 * (float)sents / w;

    printf("%f = L\n", L);
    printf("%f = S\n", S);

    grade = round(0.0588 * L - 0.296 * S - 15.8);

    return grade;
}

int main(void)
{
    string s = get_string("Text: ");

    int res = get_grade(s);

    if (res < 1){
        printf("Before Grade 1\n");
    }
    else if (res > 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %d\n", res);
    }
}

File: pset6/readability/readability.py
from cs50 import get_string
#from math import round

def grade(text):
    grade, L, S = 0.0, 0.0, 0.0
    l, w = 0, 1
    sents = 0

    for i in range(len(text)):
        c = text[i]
        if c == ' ':
            w += 1      # spaces delimit words
        elif c.isalpha():
            l += 1     # letters
        elif c == '.' or c == '!' or c == '?':
            sents += 1



    L = 100 * float(l / w)
    S = 100 * float(sents / w)

    grade = 0.0588 * L - 0.296 * S - 15.8

    return round(grade)

def main():

    while True:
        ut = get_string("Text: ")
        if len(ut) > 0:
            break

    if grade(ut) < 1:
        print("Before Grade 1")
    elif grade(ut) > 16:
        print("Grade 16+")
    else:
        print("Grade", grade(ut))


main()
